### ISSUE TITLE
Fail to generate more than one QR with different gradients

### ISSUE THREAD

**Issue by j3j5** [Bacon/BaconQrCode#186]
Generating more than one QR with different gradients while using the SVG backend causes all the QRs to render the same gradient as the first QR.

Example code:

```php
$types = ['HORIZONTAL', 'VERTICAL'];
foreach ($types as $type) {
    $gradient = new Gradient(
        new Rgb(0, 0, 0),
        new Rgb(255, 0, 0),
        GradientType::$type()
    );
    $renderer = new ImageRenderer(
        new RendererStyle(size: 400, fill: Fill::withForegroundGradient(
            new Rgb(255, 255, 255), 
            $gradient, 
            EyeFill::inherit(), 
            EyeFill::inherit(), 
            EyeFill::inherit())
        ),
        new SvgImageBackEnd()
    );
    $writer = new Writer($renderer);

    $strings[] = $writer->writeString('Hello World!');
}
```

Result:
![Screenshot_20240928_001125](https://github.com/user-attachments/assets/4ac62c98-2c08-4994-bbfc-3cb70f1aa844)

Expected result:
![Screenshot_20240928_001231](https://github.com/user-attachments/assets/3aadb608-7afc-4656-84c3-a83a3dc1031c)

The issue comes from the SVGImageBackend class, when generating the gradient, [it gives it an id based on the number of gradients present on the same QR](https://github.com/Bacon/BaconQrCode/blob/main/src/Renderer/Image/SvgImageBackEnd.php#L319), but if more than one QR is on the same page, the id is the same (`g1` or whatever the number).

There can be several ways to fix this, the easiest I can think of is to attach a random number to the id to make sure it is unique (`$id = sprintf('g%d-' . random_int(PHP_INT_MIN, PHP_INT_MAX), ++$this->gradientCount) `), another way could be creating a hash based on the gradient properties (type, start color and end color) and attach it to the id after the count. Something like:
```php
$hashValue = $this->getColorString($startColor) . $this->getColorString($endColor) . $gradient->getType();
if ($startColor instanceof Alpha) {
    $hashValue .= (string) $startColor->getAlpha();
}
$id = sprintf('g%d-' . hash('xxh128', $hashValue), ++$this->gradientCount);
```

`XXH128` is supposed to be really fast and 8.1 is now the minimum supported version which added support for it.

I'm happy to create a PR in order to fix this, let me know which way you prefer and I'll implement it.

Ps.- Thanks for this great library! :muscle:

**Comment by DASPRiD** [comment]
I think going with `random_int` might be preferred here, though preferably encoded as hex or even base64 for size reasons (basically a nano ID). Reason which speaks for that over a counting variable:

Consider generating two different QR codes in two different requests but storing them and displaying them again inline – they'd have the same ID if using an instance counter per request.

**Comment by j3j5** [comment]
I just did some tests:

```php
> hash('xxh64', 'test')
= "4fdcca5ddb678139"

hash('xxh128', 'test')
// = "6c78e0e3bd51d358d01e758642b85fb8"

dechex(PHP_INT_MAX)
// = "7fffffffffffffff"

base64_encode(PHP_INT_MAX)
// = "OTIyMzM3MjAzNjg1NDc3NTgwNw=="
```

If you are concerned about size, hex seems to be the way to go (although `xxh64` is the same size),  base64 and `xxh128` are similar in size. The only reason for going with the hash is that we get rid of any (admittedly small) possibility of id collisions (they must be the same gradient to collide, which won't matter anyway), but I agree the chances are really small and it may not be worth it the extra complexity.

As I said, up to you, I'll push PR with your choice, just lmk!

**Comment by j3j5** [comment]
Just as a curiosity, but I've done another (unscientific) test to try performance:

```php
// hash xxh64
> $test = ['abc', 'cba', 'test', 'test1', 'abcdefghijklmnopqrstuv']; 
$start = microtime(true); 
foreach(range(1, 4000000) as $i) { 
    hash('xxh64', $test[$i%5]); 
} 
$end = microtime(true); 
echo ($end - $start);
// 0.93287110328674


// hash xxh128
$test = ['abc', 'cba', 'test', 'test1', 'abcdefghijklmnopqrstuv']; 
$start = microtime(true); 
foreach(range(1, 4000000) as $i) { 
    hash('xxh128', $test[$i%5]); 
} 
$end = microtime(true); 
echo ($end - $start);
// 1.3112859725952


// dechex(random_int())
$start = microtime(true); 
foreach(range(1, 4000000) as $i) { 
    dechex(random_int(PHP_INT_MIN, PHP_INT_MAX)); 
} 
$end = microtime(true); 
echo ($end - $start);

// 6.8923921585083
```

**Comment by DASPRiD** [comment]
xxh64 should be decent enough I'd say :)

**Comment by vlakoff** [comment]
I was worried that the hash-based fix could still cause incorrect rendering in a specific scenario: two QR codes with the **same gradient** (same type + same colors → same hash → same ID) but **different sizes** embedded inline in the same HTML page. My concern was that the browser would reuse the first gradient definition, whose absolute coordinates wouldn't match the second, larger QR.

I tested it — it actually renders correctly. Looking at the generated SVG, the gradient coordinates are always in **QR module space** (e.g. `x1="0" x2="21"`), regardless of the image size. The actual pixel scaling is handled by a `transform="scale(...)"` on the parent `<g>` element.

Because `gradientUnits="userSpaceOnUse"` resolves coordinates in the **referencing element's coordinate system** (not where the gradient is defined in the DOM), the browser automatically applies the correct scale when rendering — even when reusing a "stolen" gradient definition from another SVG. So the same `x2="21"` means different pixel extents depending on which scaled context references it.

Bottom line: the collision in this same-gradient/different-size case is harmless. The fix is solid. 👍
